--- /usr/lib/urxvt/perl/tabbed.orig	2011-05-09 21:33:14.000000000 -0400
+++ /usr/lib/urxvt/perl/tabbed	2011-08-18 00:03:32.000000000 -0400
@@ -8,17 +8,20 @@
    my $text = " " x $ncol;
    my $rend = [($self->{rs_tabbar}) x $ncol];
 
+   ## github.com/dbb
+   ##
+   ## This patch changes the "new tab button" to a simple, space-padded plus sign
+
    my @ofs;
 
-   substr $text, 0, 7, "[NEW] |";
-   @$rend[0 .. 5] = ($self->{rs_tab}) x 6;
-   push @ofs, [0, 6, sub { $_[0]->new_tab }];
+   substr $text, 0, 4, " + |";
+   @$rend[0 .. 2] = ($self->{rs_tab}) x 3;
+   push @ofs, [0, 3, sub { $_[0]->new_tab }];
 
-   my $ofs = 7;
+   my $ofs = 4;
    my $idx = 0;
 
    for my $tab (@{ $self->{tabs} }) {
-      $idx++;
 
       my $act = $tab->{activity} && $tab != $self->{cur}
                 ? "*" : " ";
@@ -33,6 +36,9 @@
       push @ofs, [ $ofs, $ofs + $len, sub { $_[0]->make_current ($tab) } ];
 
       $ofs += $len + 1;
+	  ## move the postincrement down here to start tab numbers at 0
+      $idx++;
+	  ## end of changes by tabbed.patch
    }
 
    $self->{tabofs} = \@ofs;
@@ -41,7 +47,7 @@
    $self->ROW_r (0, $rend, 0, 0, $ncol);
 
    $self->want_refresh;
-}
+} ## end sub refresh
 
 sub new_tab {
    my ($self, @argv) = @_;
@@ -70,7 +76,7 @@
       -embed => $self->parent,
       @argv,
    ;
-}
+} ## end sub new_tab
 
 sub configure {
    my ($self) = @_;
@@ -88,7 +94,7 @@
       0, $self->{tabheight},
       $self->width, $self->height - $self->{tabheight}
    );
-}
+} ## end sub configure
 
 sub on_resize_all_windows {
    my ($self, $width, $height) = @_;
@@ -128,7 +134,7 @@
 
    # pass 2, delete all extraneous properties
    $self->XDeleteProperty ($self->parent, $_) for keys %$current;
-}
+} ## end sub copy_properties
 
 sub make_current {
    my ($self, $tab) = @_;
@@ -152,7 +158,7 @@
    $self->refresh;
 
    ()
-}
+} ## end sub make current
 
 sub on_focus_in {
    my ($self, $event) = @_;
@@ -202,7 +208,7 @@
    }
 
    1
-}
+} ## end sub on_button_release
 
 sub on_motion_notify {
    1
@@ -233,7 +239,7 @@
    $self->{rs_tab}    = urxvt::SET_COLOR (urxvt::DEFAULT_RSTYLE, $tabfg + 2, $tabbg + 2);
 
    ()
-}
+} ## end sub on_init
 
 sub on_start {
    my ($self) = @_;
@@ -251,7 +257,7 @@
    $self->new_tab (@argv);
 
    ()
-}
+} ## end on_start
 
 sub on_configure_notify {
    my ($self, $event) = @_;
@@ -281,7 +287,7 @@
    $self->make_current ($tab);
 
    ()
-}
+} ## end sub tab_start
 
 sub tab_destroy {
    my ($self, $tab) = @_;
@@ -301,7 +307,7 @@
    }
 
    ()
-}
+} ## end sub tab_destroy
 
 sub tab_key_press {
    my ($self, $tab, $event, $keysym, $str) = @_;
@@ -337,7 +343,7 @@
    }
 
    ()
-}
+} ## end sub tab_key_press
 
 sub tab_property_notify {
    my ($self, $tab, $event) = @_;
