--- tabbed.orig	2013-10-30 06:17:56.000000000 -0400
+++ tabbed	2014-04-02 15:48:38.271035190 -0400
@@ -44,17 +44,20 @@
    my $text = " " x $ncol;
    my $rend = [($self->{rs_tabbar}) x $ncol];
 
+   ## github.com/dbb
+   ##
+   ## This patch changes the "new tab button" to a simple, space-padded plus sign
+
    my @ofs;
 
-   substr $text, 0, 7, "[NEW] |";
-   @$rend[0 .. 5] = ($self->{rs_tab}) x 6;
-   push @ofs, [0, 6, sub { $_[0]->new_tab }];
+   substr $text, 0, 4, " + |";
+   @$rend[0 .. 2] = ($self->{rs_tab}) x 3;
+   push @ofs, [0, 3, sub { $_[0]->new_tab }];
 
-   my $ofs = 7;
+   my $ofs = 4;
    my $idx = 0;
 
    for my $tab (@{ $self->{tabs} }) {
-      $idx++;
 
       my $act = $tab->{activity} && $tab != $self->{cur}
                 ? "*" : " ";
@@ -69,6 +72,9 @@
       push @ofs, [ $ofs, $ofs + $len, sub { $_[0]->make_current ($tab) } ];
 
       $ofs += $len + 1;
+	  ## move the postincrement down here to start tab numbers at 0
+      $idx++;
+	  ## end of changes by tabbed.patch
    }
 
    $self->{tabofs} = \@ofs;
@@ -77,7 +83,7 @@
    $self->ROW_r (0, $rend, 0, 0, $ncol);
 
    $self->want_refresh;
-}
+} ## end sub refresh
 
 sub new_tab {
    my ($self, @argv) = @_;
@@ -104,7 +110,7 @@
       -embed => $self->parent,
       @argv,
    ;
-}
+} ## end sub new_tab
 
 sub configure {
    my ($self) = @_;
@@ -122,7 +128,7 @@
       0, $self->{tabheight},
       $self->width, $self->height - $self->{tabheight}
    );
-}
+} ## end sub configure
 
 sub on_resize_all_windows {
    my ($self, $width, $height) = @_;
@@ -162,7 +168,7 @@
 
    # pass 2, delete all extraneous properties
    $self->XDeleteProperty ($self->parent, $_) for keys %$current;
-}
+} ## end sub copy_properties
 
 sub make_current {
    my ($self, $tab) = @_;
@@ -186,7 +192,7 @@
    $self->refresh;
 
    ()
-}
+} ## end sub make current
 
 sub on_focus_in {
    my ($self, $event) = @_;
@@ -244,7 +250,7 @@
    }
 
    1
-}
+} ## end sub on_button_release
 
 sub on_motion_notify {
    1
@@ -275,7 +281,7 @@
    $self->{rs_tab}    = urxvt::SET_COLOR (urxvt::DEFAULT_RSTYLE, $tabfg + 2, $tabbg + 2);
 
    ()
-}
+} ## end sub on_init
 
 sub on_start {
    my ($self) = @_;
@@ -293,7 +299,7 @@
    $self->new_tab (@argv);
 
    ()
-}
+} ## end on_start
 
 sub on_configure_notify {
    my ($self, $event) = @_;
@@ -323,7 +329,7 @@
    $self->make_current ($tab);
 
    ()
-}
+} ## end sub tab_start
 
 sub tab_destroy {
    my ($self, $tab) = @_;
@@ -343,7 +349,7 @@
    }
 
    ()
-}
+} ## end sub tab_destroy
 
 sub tab_key_press {
    my ($self, $tab, $event, $keysym, $str) = @_;
@@ -379,7 +385,7 @@
    }
 
    ()
-}
+} ## end sub tab_key_press
 
 sub tab_property_notify {
    my ($self, $tab, $event) = @_;
